### PR TITLE
feat: add api to manage Ingredients

### PR DESCRIPTION
**Pull request by OmriBarZik**
Added new endpoints to the site

* add single Ingredient
* update single Ingredient
* remove single Ingredient
* get single Ingredient
* get multiple Ingredient

Added Zod for schema validations

### CODE REVIEW

File: lib/ingredient.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  return getIngredient(insertData.insertedId.toString());
+}
+
+export async function updateIngredient(
+  id: string,
+  ingredient: Partial<Ingredient>
+) {
+  const collection = await getCollection();
+
+  const updatedIngredient = await collection.findOneAndUpdate(
```
Comment (InbarDanieli) [inline lib/ingredient.ts:31]: what will happen if this id doesn't exist?

File: lib/ingredient.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  return updatedIngredient.value;
+}
+
+export async function getIngredient(id: string) {
+  const collection = await getCollection();
+
+  return collection.findOne({ _id: new ObjectId(id) });
+}
+
+export async function getIngredients(): Promise<Ingredient[]> {
```
Comment (InbarDanieli) [inline lib/ingredient.ts:46]: did you add type to all the promises?

**Review by InbarDanieli — CHANGES_REQUESTED** [review]

File: lib/ingredient.ts (hunk 1 above)
Comment (OmriBarZik) [reply lib/ingredient.ts:33]: It will return an null of no object is found

File: lib/ingredient.ts (hunk 2 above)
Comment (OmriBarZik) [reply lib/ingredient.ts:48]: Done

**Review by InbarDanieli — APPROVED** [review]

**Pull request merged**